pkg/pillar/cmd/zedrouter/appnumonunet.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Network Instance/underlay network IP Address Management,
+// App Number management Module
+
+package zedrouter
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// Network Instance Application number/IP Address Management
+// ---------------------------------------------------------
+//  AppNum  Module (bitMap & persistent AppNumMap)
+//   - uNetAppNumMapGet()
+//   - uNetAppNumGet()
+//   - uNetAppNumAlloc()
+//   - uNetAppNumDelete()
+//   - networkInstanceAppNumMapInitialize()
+//   - networkInstanceAppNumMapDelete()
+//   - getDhcpAppNumCountOnDhcpRange()
+//   - getDhcpAppNumCountOnSubnet()
+//   - networkInstanceDhcpAppNumCountMax()
+//  BitMap : is for the Dynamic dhcp address/app number assignments
+// For BitMap:(DhcpRange.Start - DhcpRange.End)
+//    - bitMapAppNumAdd(), set a bit in the BitMap
+//    - bitMapAppNumDelete(), reset a bit in the BitMap
+
+// mapped on NetworkInstance ID
+var netInstAppNumMap map[string]*types.NetworkInstanceAppNumMapInfo
+
+// returns the AppNumMap structure, for a network instance
+func uNetAppNumMapGet(networkID string) *types.NetworkInstanceAppNumMapInfo {
+	if netInstAppNumMap != nil {
+		if appNumMapInfo, ok := netInstAppNumMap[networkID]; ok {
+			if appNumMapInfo.DhcpAppNumCount != 0 {
+				return appNumMapInfo
+			}
+		}
+	}
+	return nil
+}
+
+// Get the AppNum for a bridge or, an app instance with static IP
+func uNetAppNumGet(ctx *zedrouterContext, networkID string, appID string,
+	ipAddr net.IP, mac net.HardwareAddr) (int, error) {
+
+	appNumMapInfo := uNetAppNumMapGet(networkID)
+	if appNumMapInfo == nil || ipAddr == nil {
+		return types.AppNumInvalid, nil
```

Comment:
Since the caller don't check for AppNumInvalid, please return an error instead and remove the AppNumInvalid definition.